src/components/map-tephra-thickness-layer.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from "react";
+import * as Leaflet from "leaflet";
+import * as Color from "color";
+import * as d3 from "d3";
+import { inject, observer } from "mobx-react";
+import { BaseComponent } from "./base";
+import gridTephraCalc from "../tephra2";
+import { Ipoint } from "../interfaces";
+import { LayerGroup, GeoJSON } from "react-leaflet";
+import { LatLngToLocal } from "./coordinateSpaceConversion";
+import { MultiPolygon } from "geojson";
+
+interface IProps {
+    corner1Bound: Leaflet.LatLng;
+    corner2Bound: Leaflet.LatLng;
+    viewportBounds: Leaflet.LatLngBounds;
+    volcanoPos: Leaflet.LatLng;
+    gridSize: number;
+    map: Leaflet.Map | null;
+    windSpeed: number;
+    windDirection: number;
+    colHeight: number;
+    mass: number;
+    particleSize: number;
+}
+
+interface IState {}
+
+@inject("stores")
+@observer
+export class MapTephraThicknessLayer extends BaseComponent<IProps, IState> {
+
+    // Not ideal, but geoJson does not update with state changes
```

Comment:
Interesting. This shouldn't be necessary.  Figuring out why a component isn't updating based on its properties can be tricky.

Each `<GeoJSON>` component in the collection  needs to have a unique `key` property, but these keys are usually simple indexes (something like  1- 8 in this example I think).

Any way, this isn't a big deal, and we can return to it later when we are polishing up.